core/src/main/java/org/jdbi/v3/core/mapper/reflect/ConstructorInstanceFactory.java

```diff
@@ -13,38 +13,50 @@
  */
 package org.jdbi.v3.core.mapper.reflect;
 
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
 import java.lang.reflect.Constructor;
 import java.lang.reflect.Field;
 import java.lang.reflect.Modifier;
 import java.lang.reflect.Parameter;
 import java.lang.reflect.Type;
 import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.WeakHashMap;
 import java.util.function.Supplier;
 import java.util.stream.Stream;
 
 import org.jdbi.v3.core.internal.exceptions.Unchecked;
 
+import static java.util.Collections.synchronizedMap;
 import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toUnmodifiableList;
 
 class ConstructorInstanceFactory<T> extends InstanceFactory<T> {
-    private final Constructor<T> constructor;
+    private static final MethodHandles.Lookup LOOKUP = MethodHandles.lookup();
```

Comment:
lmk if this shouldn't be cached.

Reply:
Let me know if you want the weak hashmap added back vs caching in the constructor instance factory (saved the previous version, so I can quickly update).

Reply:
So I think there is a bit of a misunderstanding here. I didn't mean to cache the lookup factory instance itself, but the `constructorHandle` instance itself.

Something like `SynchronizedMap<WeakHashMap<ConstructorHandleAndTypes, MethodHandle>>>` (referring to the key type from the other PR)

Sorry I wasn't clear, and sorry for taking so long to respond! Been a busy couple weeks at work.